vscode-lean4/src/utils/leanInstaller.ts

```diff
@@ -331,34 +328,40 @@ export class LeanInstaller implements Disposable {
             // user is requesting an explicit version!
             options = ['+' + version, '--version']
         }
+
+        const result : LeanVersion = { version: '', error: undefined }
         try {
             // If folderPath is undefined, this will use the process environment for cwd.
             // Specifically, if the extension was not opened inside of a folder, it
             // looks for a global (default) installation of Lean. This way, we can support
             // single file editing.
+            console.log(`executeWithProgress ${cmd} ${options}`)
             const stdout = await this.executeWithProgress('Checking Lean setup...', cmd, options, folderPath)
             if (!stdout) {
-                return { version: '', error: 'lean not found' };
+                result.error = 'lean not found'
             }
```

Comment:
I think this conditional is redundant: if we have `!stdout`, we will also have `!match` and so we `return` below without actually using `result`

Reply:
the diff is confusing, it is now an `if (!stdout) { ... } else if (stdout.indexOf('no default toolchain') > 0) { ... } else { ... }` and the version matching is in the last else, so there is no redundant conditional.

Reply:
Alright, but then the inner `!stdout` conditional is redundant